scripts/server.js

```diff
@@ -171,11 +174,16 @@ async function main() {
   router.post('/posters', async ctx => {
     const { buildId, component, props, template } = ctx.request.body;
     const posters = [];
+
     for (let i = 0; i < props.length; i++) {
-      // eslint-disable-next-line no-await-in-loop
-      posters.push(await generatePoster(buildId, component, template, props[i]));
+      posters.push(generatePoster(buildId, component, template, props[i], i));
+    }
+
+    try {
+      ctx.body = await Promise.all(posters);
+    } catch (err) {
+      ctx.throw(500, err.message || 'Poster generation failed.');
```

Comment:
This could probably use some more granular error catching, now it will throw if just any poster fails. But maybe that's also appropriate?

Reply:
Let's go with this for now